docs/List.md

```diff
@@ -772,6 +765,304 @@ export PostList;
+        <TextInput label="Title" source="title" defaultValue="Hello, World!" />
+    </Filter>
+);
+```
+
+Children of the `<Filter>` form are regular inputs. `<Filter>` hides them all by default, except those that have the `alwaysOn` prop.
+
+**Tip**: For technical reasons, react-admin does not accept children of `<Filter>` having both a `defaultValue` and `alwaysOn`. To set default values for always on filters, use the [`filterDefaultValues`](#filter-default-values) prop of the <List> component instead.
+
+**Tip**: Once a user sets a filter, react-admin persists the filter value in the application state, so that when the user comes back to the list, they should see the filtered list. That's a design choice.
+
+### Linking To A Pre-Filtered List
+
+As the filter values are taken from the URL, you can link to a pre-filtered list by setting the query parameter.
+
+For instance, if you have a list of tags, you can display a button for each category to link to the list of posts filtered by that tag:
+
+{% raw %}
+```jsx
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import { Link } from 'react-router-dom';
+import { stringify } from 'query-string';
+
+const LinkToRelatedProducts = ({ record }) => {
+    const translate = useTranslate();
+    return record ? (
+        <Button
+            color="primary"
+            component={Link}
+            to={{
+                pathname: '/posts',
+                search: stringify({
+                    page: 1,
+                    perPage: 25,
+                    sort: 'id',
+                    order: 'DESC',
+                    filter: JSON.stringify({ category_id: record.id }),
+                }),
+            }}
+        >
+            All posts with the category {record.name} ; 
+        </Button>
+    ) : null;
+};
+```
+{% endraw %}
+
+You can use this button e.g. as a child of `<Datagrid>`. You can also create a custom Menu button with that technique to link to the unfiltered list by setting the filter value to `{}`.
+
+**Tip**: You have to pass *all* the query string parameters - not just `filter`. That's a current limitation of react-admin.
+
+### Full-Text Search
+
+In addition to [the usual input types](./Inputs.md) (`<TextInput>`, `<SelectInput>`, `<ReferenceInput>`, etc.), you can use the `<SearchInput>`, which is designed especially for the filter form. It's like a `<TextInput resettable>` with a magnifier glass icon - exactly the type of input users look for when they want to do a full-text search. 
+
+![`<SearchInput>`](./img/search_input.gif)
+
+```jsx
+import { Filter, SearchInput, TextInput } from 'react-admin';
+
+const PostFilter = props => (
+    <Filter {...props}>
+        <SearchInput source="q" alwaysOn />
+    </Filter>
+);
+```
+
+In the example given above, the `q` filter triggers a full-text search on all fields. It's your responsibility to implement the full-text filtering capabilities in your API.
```

Comment:
I would probably be nice to explain the `q` filter can be handled/converted in the dataProvider as well